src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -330,7 +330,7 @@ private function resolveCommands($value)
                 throw new \LogicException('Resolving commands requires the Symfony Process component.');
             }
 
-            $process = new Process('echo '.$matches[0]);
+            $process = \method_exists(Process::class, 'withShell') ? Process::withShell('echo '.$matches[0]) : new Process('echo '.$matches[0]);
```

Comment:
What about adding a comment to not forget simplifying this piece of code again in 5.0?

Reply:
we'll have many places to clean up like this when preparing 5.0, we will spot it with the usual `grep method_exists`